persistence.js

```diff
@@ -554,7 +554,8 @@ class RedisPersistence extends CachedPersistence {
 }
 
 async function * matchRetained (db, qlobber, hasClusters) {
-  for (const key of await getRetainedKeys(db, hasClusters)) {
```

Comment:
why this change? If we make the getRetainedKeys return a stream this would be more performant as it doesn't require to load all keys in memory